### PR TITLE
Cw 12

### PR DESCRIPTION
**Pull request by Abomy**
Seeder rewrite and all tables are fill with data

### CODE REVIEW

File: foundationz_backend/db/migrations/20220404040430_allotments.cjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+exports.up = function (knex) {
+    return knex.schema.createTable("allotments", table => {
+        table.uuid("id").primary().notNullable();
+        table.uuid("booking_id")
+            .references('bookings.id');
+        table.datetime("start_date");
+        table.datetime("end_date");
+        table.string("type");
```
Comment (Kysumi) [inline foundationz_backend/db/migrations/20220404040430_allotments.cjs:8]: we should probs make this a foreign key https://foundationz.youtrack.cloud/issue/CW-16

File: foundationz_backend/db/migrations/20220404040430_allotments.cjs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }).createTable("allotment_rebook", table => {
+        table.uuid("id").primary().notNullable();
+        table.uuid("allotment_id")
```
Comment (Kysumi) [inline foundationz_backend/db/migrations/20220404040430_allotments.cjs:11]: should be not nullable

File: foundationz_backend/db/migrations/20220405032554_contacts.cjs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+exports.up = function (knex) {
+
+    return knex.schema.createTable("contacts", table => {
+        table.uuid("id").notNullable().primary();
+        table.string("first_name");
+        table.string("surname");
+        table.timestamps(false, true);
+        table.uuid("organization_id").references("organizations.id");
```
Comment (Kysumi) [inline foundationz_backend/db/migrations/20220405032554_contacts.cjs:8]: noNullable i think here as well.

File: foundationz_backend/db/seeds/functionals/baseSeeder.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+export default class BaseSeeder {
+  data = [];
+  amount = 10;
+
+  constructor(amount) {
+    if (!amount) {
+      throw "Set an amount you POS";
```
Comment (Kysumi) [inline foundationz_backend/db/seeds/functionals/baseSeeder.js:7]: 🤔 "An amount is required for the seeders constructor"

File: foundationz_backend/db/seeds/functionals/contacts.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { v4 } from "uuid";
+import { faker } from "@faker-js/faker";
+import BaseSeeder from "./baseSeeder.js";
+import { Organizations } from "../seeder.js";
+
+export default class ContactSeeder extends BaseSeeder {
+  fields() {
+    return {
+      id: v4(),
+      first_name: `${faker.name.firstName()}`,
```
Comment (Kysumi) [inline foundationz_backend/db/seeds/functionals/contacts.js:10]: the ` are not required here. 

`
       first_name: faker.name.firstName(),
`

could look like this instead

File: foundationz_backend/db/seeds/functionals/clear.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+export async function ClearDatabase(knex) {
```
Comment (Kysumi) [inline foundationz_backend/db/seeds/functionals/clear.js:1]: camelCase please

File: foundationz_backend/db/seeds/functionals/user.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { faker } from "@faker-js/faker";
+import BaseSeeder from "./baseSeeder.js";
+import { v4 } from "uuid";
+
+export default class UserSeeder extends BaseSeeder {
+  fields() {
+    return {
+      id: v4(),
+      first_name: `${faker.name.firstName()}`,
```
Comment (Kysumi) [inline foundationz_backend/db/seeds/functionals/user.js:9]: same as above

**Review by Kysumi — COMMENTED** [review]
Looks good

File: foundationz_backend/db/seeds/functionals/baseSeeder.js (hunk 4 above)
Comment (Abomy) [reply foundationz_backend/db/seeds/functionals/baseSeeder.js:7]: 😂 must have had too much sauce that night

File: foundationz_backend/db/migrations/20220404040430_allotments.cjs (hunk 1 above)
Comment (Abomy) [reply foundationz_backend/db/migrations/20220404040430_allotments.cjs:8]: Good plan, ill sort that out in that ticket

**Review by Kysumi — APPROVED** [review]

**Pull request merged**